dashboard/app/models/level.rb

```diff
@@ -349,6 +351,27 @@ def strip_name
     self.name = name.to_s.strip unless name.nil?
   end
 
+  def log_changes
+    log = JSON.parse(audit_log || "[]")
+
+    # serialize 'properties' changes
+    changed = self.changed
+    if changed.include?('properties')
+      changed.delete('properties')
+      changed_attributes['properties'].each do |key, value|
+        changed.push(key) if properties[key] != value
+      end
+    end
+
+    log.push(
+      {
+        changed_at: Time.now,
+        changed: changed
+      }
+    )
+    self.audit_log = JSON.dump(log)
```

Comment:
Ruby ignorance on my part: Why do you need `self` to refer to `audit_log` here but you don't need it when parsing the log on line 355?